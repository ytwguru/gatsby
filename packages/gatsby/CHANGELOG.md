# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="2.0.0-beta.64"></a>

# [2.0.0-beta.64](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.63...gatsby@2.0.0-beta.64) (2018-07-31)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.63"></a>

# [2.0.0-beta.63](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.62...gatsby@2.0.0-beta.63) (2018-07-31)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.62"></a>

# [2.0.0-beta.62](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.61...gatsby@2.0.0-beta.62) (2018-07-31)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.61"></a>

# [2.0.0-beta.61](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.60...gatsby@2.0.0-beta.61) (2018-07-28)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.60"></a>

# [2.0.0-beta.60](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.59...gatsby@2.0.0-beta.60) (2018-07-27)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.59"></a>

# [2.0.0-beta.59](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.58...gatsby@2.0.0-beta.59) (2018-07-25)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.58"></a>

# [2.0.0-beta.58](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.57...gatsby@2.0.0-beta.58) (2018-07-25)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.57"></a>

# [2.0.0-beta.57](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.56...gatsby@2.0.0-beta.57) (2018-07-24)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.56"></a>

# [2.0.0-beta.56](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.55...gatsby@2.0.0-beta.56) (2018-07-24)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.55"></a>

# [2.0.0-beta.55](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.54...gatsby@2.0.0-beta.55) (2018-07-21)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.54"></a>

# [2.0.0-beta.54](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.53...gatsby@2.0.0-beta.54) (2018-07-20)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.53"></a>

# [2.0.0-beta.53](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.52...gatsby@2.0.0-beta.53) (2018-07-20)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.52"></a>

# [2.0.0-beta.52](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.51...gatsby@2.0.0-beta.52) (2018-07-20)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.51"></a>

# [2.0.0-beta.51](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.50...gatsby@2.0.0-beta.51) (2018-07-20)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.50"></a>

# [2.0.0-beta.50](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.49...gatsby@2.0.0-beta.50) (2018-07-20)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.49"></a>

# [2.0.0-beta.49](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.48...gatsby@2.0.0-beta.49) (2018-07-20)

### Bug Fixes

- Bumping Serve Up to 9.2.0 Broke Calls to Serve's API ([#6598](https://github.com/gatsbyjs/gatsby/issues/6598)) ([ce9c3a8](https://github.com/gatsbyjs/gatsby/commit/ce9c3a8))

<a name="2.0.0-beta.48"></a>

# [2.0.0-beta.48](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.47...gatsby@2.0.0-beta.48) (2018-07-19)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.47"></a>

# [2.0.0-beta.47](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.46...gatsby@2.0.0-beta.47) (2018-07-18)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.46"></a>

# [2.0.0-beta.46](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.45...gatsby@2.0.0-beta.46) (2018-07-18)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.45"></a>

# [2.0.0-beta.45](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.44...gatsby@2.0.0-beta.45) (2018-07-18)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.44"></a>

# [2.0.0-beta.44](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.43...gatsby@2.0.0-beta.44) (2018-07-18)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.43"></a>

# [2.0.0-beta.43](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.42...gatsby@2.0.0-beta.43) (2018-07-17)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.42"></a>

# [2.0.0-beta.42](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.41...gatsby@2.0.0-beta.42) (2018-07-17)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.41"></a>

# [2.0.0-beta.41](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.40...gatsby@2.0.0-beta.41) (2018-07-17)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.40"></a>

# [2.0.0-beta.40](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.39...gatsby@2.0.0-beta.40) (2018-07-16)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.39"></a>

# [2.0.0-beta.39](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.38...gatsby@2.0.0-beta.39) (2018-07-16)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.38"></a>

# [2.0.0-beta.38](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.37...gatsby@2.0.0-beta.38) (2018-07-15)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.37"></a>

# [2.0.0-beta.37](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.36...gatsby@2.0.0-beta.37) (2018-07-15)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.36"></a>

# [2.0.0-beta.36](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.35...gatsby@2.0.0-beta.36) (2018-07-14)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.35"></a>

# [2.0.0-beta.35](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.34...gatsby@2.0.0-beta.35) (2018-07-13)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.34"></a>

# [2.0.0-beta.34](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.33...gatsby@2.0.0-beta.34) (2018-07-13)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.33"></a>

# [2.0.0-beta.33](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.32...gatsby@2.0.0-beta.33) (2018-07-13)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.32"></a>

# [2.0.0-beta.32](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.31...gatsby@2.0.0-beta.32) (2018-07-13)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.31"></a>

# [2.0.0-beta.31](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.30...gatsby@2.0.0-beta.31) (2018-07-13)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.30"></a>

# [2.0.0-beta.30](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.29...gatsby@2.0.0-beta.30) (2018-07-13)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.29"></a>

# [2.0.0-beta.29](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.28...gatsby@2.0.0-beta.29) (2018-07-12)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.28"></a>

# [2.0.0-beta.28](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.27...gatsby@2.0.0-beta.28) (2018-07-12)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.27"></a>

# [2.0.0-beta.27](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.26...gatsby@2.0.0-beta.27) (2018-07-12)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.26"></a>

# [2.0.0-beta.26](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.25...gatsby@2.0.0-beta.26) (2018-07-12)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.25"></a>

# [2.0.0-beta.25](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.24...gatsby@2.0.0-beta.25) (2018-07-11)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.24"></a>

# [2.0.0-beta.24](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.23...gatsby@2.0.0-beta.24) (2018-07-11)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.23"></a>

# [2.0.0-beta.23](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.22...gatsby@2.0.0-beta.23) (2018-07-11)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.22"></a>

# [2.0.0-beta.22](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.21...gatsby@2.0.0-beta.22) (2018-07-11)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.21"></a>

# [2.0.0-beta.21](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.20...gatsby@2.0.0-beta.21) (2018-07-11)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.20"></a>

# [2.0.0-beta.20](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.19...gatsby@2.0.0-beta.20) (2018-07-09)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.19"></a>

# [2.0.0-beta.19](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.18...gatsby@2.0.0-beta.19) (2018-07-06)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.18"></a>

# [2.0.0-beta.18](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.17...gatsby@2.0.0-beta.18) (2018-07-06)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.17"></a>

# [2.0.0-beta.17](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.16...gatsby@2.0.0-beta.17) (2018-07-03)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.16"></a>

# [2.0.0-beta.16](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.15...gatsby@2.0.0-beta.16) (2018-07-03)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.15"></a>

# [2.0.0-beta.15](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.14...gatsby@2.0.0-beta.15) (2018-07-02)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.14"></a>

# [2.0.0-beta.14](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.13...gatsby@2.0.0-beta.14) (2018-07-02)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.13"></a>

# [2.0.0-beta.13](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.12...gatsby@2.0.0-beta.13) (2018-06-29)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.12"></a>

# [2.0.0-beta.12](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.11...gatsby@2.0.0-beta.12) (2018-06-26)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.11"></a>

# [2.0.0-beta.11](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.10...gatsby@2.0.0-beta.11) (2018-06-25)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.10"></a>

# [2.0.0-beta.10](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.9...gatsby@2.0.0-beta.10) (2018-06-25)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.9"></a>

# [2.0.0-beta.9](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.8...gatsby@2.0.0-beta.9) (2018-06-21)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.8"></a>

# [2.0.0-beta.8](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.7...gatsby@2.0.0-beta.8) (2018-06-21)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.7"></a>

# [2.0.0-beta.7](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.6...gatsby@2.0.0-beta.7) (2018-06-21)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.6"></a>

# [2.0.0-beta.6](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.5...gatsby@2.0.0-beta.6) (2018-06-21)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.5"></a>

# [2.0.0-beta.5](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.4...gatsby@2.0.0-beta.5) (2018-06-21)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.4"></a>

# [2.0.0-beta.4](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.3...gatsby@2.0.0-beta.4) (2018-06-20)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.3"></a>

# [2.0.0-beta.3](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.2...gatsby@2.0.0-beta.3) (2018-06-20)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.2"></a>

# [2.0.0-beta.2](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.1...gatsby@2.0.0-beta.2) (2018-06-19)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.1"></a>

# [2.0.0-beta.1](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.0.0-beta.0...gatsby@2.0.0-beta.1) (2018-06-17)

**Note:** Version bump only for package gatsby

<a name="2.0.0-beta.0"></a>

# [2.0.0-beta.0](https://github.com/gatsbyjs/gatsby/compare/gatsby@1.9.272...gatsby@2.0.0-beta.0) (2018-06-17)

**Note:** Version bump only for package gatsby
