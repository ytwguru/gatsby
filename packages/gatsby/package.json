{
  "name": "gatsby",
  "description": "React.js Static Site Generator",
  "version": "2.0.0-beta.64",
  "author": "Kyle Mathews <mathews.kyle@gmail.com>",
  "bin": {
    "gatsby": "./dist/bin/gatsby.js"
  },
  "bugs": {
    "url": "https://github.com/gatsbyjs/gatsby/issues"
  },
  "dependencies": {
    "@babel/code-frame": "7.0.0-beta.52",
    "@babel/core": "7.0.0-beta.52",
    "@babel/parser": "7.0.0-beta.52",
    "@babel/plugin-proposal-class-properties": "7.0.0-beta.52",
    "@babel/plugin-syntax-dynamic-import": "7.0.0-beta.52",
    "@babel/plugin-transform-runtime": "7.0.0-beta.52",
    "@babel/polyfill": "7.0.0-beta.52",
    "@babel/preset-env": "7.0.0-beta.52",
    "@babel/preset-flow": "7.0.0-beta.52",
    "@babel/preset-react": "7.0.0-beta.52",
    "@babel/runtime": "7.0.0-beta.52",
    "@babel/traverse": "7.0.0-beta.52",
    "async": "^2.1.2",
    "autoprefixer": "^8.6.5",
    "babel-core": "7.0.0-bridge.0",
    "babel-eslint": "^8.2.2",
    "babel-loader": "8.0.0-beta.4",
    "babel-plugin-add-module-exports": "^0.2.1",
    "babel-plugin-dynamic-import-node": "^1.2.0",
    "babel-plugin-remove-graphql-queries": "^2.0.2-beta.8",
    "better-queue": "^3.8.6",
    "bluebird": "^3.5.0",
    "chalk": "^2.3.2",
    "chokidar": "^2.0.2",
    "common-tags": "^1.4.0",
    "convert-hrtime": "^2.0.0",
    "copyfiles": "^1.2.0",
    "core-js": "^2.5.0",
    "css-loader": "^1.0.0",
    "cssnano": "^4.0.2",
    "debug": "^3.1.0",
    "del": "^3.0.0",
    "detect-port": "^1.2.1",
    "devcert-san": "^0.3.3",
    "domready": "^1.0.8",
    "dotenv": "^4.0.0",
    "eslint": "^4.19.1",
    "eslint-config-react-app": "3.0.0-next.66cc7a90",
    "eslint-loader": "^2.0.0",
    "eslint-plugin-flowtype": "^2.46.1",
    "eslint-plugin-graphql": "^2.0.0",
    "eslint-plugin-import": "^2.9.0",
    "eslint-plugin-jsx-a11y": "^6.0.3",
    "eslint-plugin-react": "^7.8.2",
    "express": "^4.16.3",
    "express-graphql": "^0.6.12",
    "fast-levenshtein": "~2.0.4",
    "file-loader": "^1.1.11",
    "flat": "^4.0.0",
    "friendly-errors-webpack-plugin": "^1.6.1",
    "front-matter": "^2.1.0",
    "fs-extra": "^5.0.0",
    "gatsby-cli": "^2.0.0-beta.7",
    "gatsby-link": "^2.0.0-beta.5",
    "gatsby-plugin-page-creator": "^2.0.0-beta.5",
    "gatsby-react-router-scroll": "^2.0.0-beta.3",
    "glob": "^7.1.1",
    "graphql": "^0.13.2",
    "graphql-relay": "^0.5.5",
    "graphql-skip-limit": "^2.0.0-beta.3",
    "graphql-type-json": "^0.2.1",
    "hash-mod": "^0.0.5",
    "history": "^4.6.2",
    "invariant": "^2.2.4",
    "is-relative": "^1.0.0",
    "is-relative-url": "^2.0.0",
    "jest-worker": "^23.2.0",
    "joi": "12.x.x",
    "json-loader": "^0.5.7",
    "json-stringify-safe": "^5.0.1",
    "json5": "^0.5.0",
    "kebab-hash": "^0.1.2",
    "lodash": "^4.17.4",
    "lodash-id": "^0.14.0",
    "lowdb": "0.16.2",
    "md5": "^2.2.1",
    "md5-file": "^3.1.1",
    "mime": "^2.2.0",
    "mini-css-extract-plugin": "^0.4.0",
    "mitt": "^1.1.2",
    "mkdirp": "^0.5.1",
    "moment": "^2.21.0",
    "name-all-modules-plugin": "^1.0.1",
    "node-libs-browser": "^2.0.0",
    "normalize-path": "^2.1.1",
    "null-loader": "^0.1.1",
    "opentracing": "^0.14.3",
    "opn": "^5.3.0",
    "parse-filepath": "^1.0.1",
    "path-exists": "^3.0.0",
    "physical-cpu-count": "^2.0.0",
    "postcss-flexbugs-fixes": "^3.0.0",
    "postcss-loader": "^2.1.3",
    "raw-loader": "^0.5.1",
    "react-dev-utils": "^4.2.1",
    "react-error-overlay": "^3.0.0",
    "react-hot-loader": "^4.1.0",
    "react-lifecycles-compat": "^3.0.2",
    "react-router": "^4.1.1",
    "react-router-dom": "^4.1.1",
    "redux": "^3.6.0",
    "relay-compiler": "1.5.0",
    "request": "^2.85.0",
    "serve": "^9.2.0",
    "shallow-compare": "^1.2.2",
    "sift": "^5.1.0",
    "signal-exit": "^3.0.2",
    "slash": "^1.0.0",
    "socket.io": "^2.0.3",
    "string-similarity": "^1.2.0",
    "style-loader": "^0.21.0",
    "treeify": "^1.1.0",
    "type-of": "^2.0.1",
    "uglifyjs-webpack-plugin": "^1.2.4",
    "url-loader": "^1.0.1",
    "uuid": "^3.1.0",
    "v8-compile-cache": "^1.1.0",
    "webpack": "^4.12.0",
    "webpack-dev-middleware": "^3.0.1",
    "webpack-dev-server": "^3.1.1",
    "webpack-hot-middleware": "^2.21.0",
    "webpack-md5-hash": "^0.0.6",
    "webpack-merge": "^4.1.0",
    "webpack-stats-plugin": "^0.1.5",
    "yaml-loader": "^0.5.0"
  },
  "devDependencies": {
    "@babel/cli": "7.0.0-beta.52",
    "@babel/plugin-proposal-class-properties": "7.0.0-beta.52",
    "@babel/plugin-transform-runtime": "7.0.0-beta.52",
    "@babel/preset-env": "7.0.0-beta.52",
    "@babel/preset-flow": "7.0.0-beta.52",
    "@babel/preset-react": "7.0.0-beta.52",
    "@babel/runtime": "7.0.0-beta.52",
    "cross-env": "^5.1.4",
    "jest": "^22.4.3",
    "lerna": "^2.9.0",
    "nyc": "^7.0.0",
    "rimraf": "^2.6.1"
  },
  "engines": {
    "node": ">6.0.0"
  },
  "files": [
    "./cache-dir",
    "./dist",
    "graphql.js",
    "index.d.ts"
  ],
  "homepage": "https://github.com/gatsbyjs/gatsby#readme",
  "keywords": [
    "blog",
    "generator",
    "jekyll",
    "markdown",
    "react",
    "ssg",
    "website"
  ],
  "license": "MIT",
  "main": "cache-dir/gatsby-browser-entry.js",
  "peerDependencies": {
    "react": "^16.2.0",
    "react-dom": "^16.2.0"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/gatsbyjs/gatsby.git"
  },
  "resolutions": {
    "graphql": "^0.13.2"
  },
  "scripts": {
    "build": "rimraf dist && npm run build:src && npm run build:internal-plugins && npm run build:rawfiles",
    "build:internal-plugins": "copyfiles -u 1 src/internal-plugins/**/package.json dist",
    "build:rawfiles": "copyfiles -u 1 src/internal-plugins/**/raw_* dist",
    "build:src": "babel src --out-dir dist --source-maps --ignore **/gatsby-cli.js,**/raw_*,**/__tests__",
    "clean-test-bundles": "find test/ -type f -name bundle.js* -exec rm -rf {} +",
    "prepare": "cross-env NODE_ENV=production npm run build",
    "test-coverage": "node_modules/.bin/nyc --reporter=lcov --reporter=text npm test",
    "watch": "rimraf dist && mkdir dist && npm run build:internal-plugins && npm run build:rawfiles && npm run build:src -- --watch"
  },
  "types": "index.d.ts",
  "gitHead": "788133e56dd005663f353fff90ffa37ce2fd87b9"
}
