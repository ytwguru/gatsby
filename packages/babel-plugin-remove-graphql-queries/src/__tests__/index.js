const babel = require(`babel-core`)
const reactPreset = require(`@babel/preset-react`)
const plugin = require(`../`)

function matchesSnapshot(query) {
  const { code } = babel.transform(query, {
    presets: [reactPreset],
    plugins: [plugin],
  })
  expect(code).toMatchSnapshot()
}

it(`Transforms queries in <StaticQuery>`, () => {
  matchesSnapshot(`
  import React from 'react'
  import { graphql, StaticQuery } from 'gatsby'

  export default () => (
    <StaticQuery
      query={graphql\`{site { siteMetadata { title }}}\`}
      render={data => <div>{data.site.siteMetadata.title}</div>}
    />
  )
  `)
})

it(`Transforms queries in page components`, () => {
  matchesSnapshot(`
  import { graphql } from 'gatsby'

  export const query = graphql\`
     {
       site { siteMetadata { title }}
     }
  \`
  `)
})

it(`allows the global tag`, () => {
  matchesSnapshot(
    `
  export const query = graphql\`
     {
       site { siteMetadata { title }}
     }
  \`
  `
  )
})

it(`distinguishes between the right tags`, () => {
  matchesSnapshot(
    `
  const foo = styled('div')\`
     {
       $\{foo}
     }
  \`

  const pulse = keyframes\`
    0% {
      transform: scale(1);
      animation-timing-function: ease-in;
    }
    25% {
      animation-timing-function: ease-out;
      transform: scale(1.05);
    }
    50% {
      transform: scale(1.12);
      animation-timing-function: ease-in;
    }
    to {
      transform: scale(1);
      animation-timing-function: ease-out;
    }
  \`;


  export const query = graphql\`
     {
       site { siteMetadata { title }}
     }
  \`
  `
  )
})

it(`handles import aliasing`, () => {
  matchesSnapshot(
    `
  import { graphql as gql } from 'gatsby'

  export const query = gql\`
     {
       site { siteMetadata { title }}
     }
  \`
  `
  )
})

it(`handles require`, () => {
  matchesSnapshot(
    `
  const { graphql } = require('gatsby')

  export const query = graphql\`
     {
       site { siteMetadata { title }}
     }
  \`
  `
  )
})

it(`handles require namespace`, () => {
  matchesSnapshot(
    `
  const Gatsby = require('gatsby')

  export const query = Gatsby.graphql\`
     {
       site { siteMetadata { title }}
     }
  \`
  `
  )
})
it(`handles require alias`, () => {
  matchesSnapshot(
    `
  const { graphql: gql } = require('gatsby')

  export const query = gql\`
     {
       site { siteMetadata { title }}
     }
  \`
  `
  )
})

it(`Leaves other graphql tags alone`, () => {
  matchesSnapshot(
    `
  import React from 'react'
  import { graphql } from 'relay'

  export default () => (
    <div>{data.site.siteMetadata.title}</div>
  )

  export const query = graphql\`
     {
       site { siteMetadata { title }}
     }
  \`
  `
  )
})

it(`Removes all gatsby queries`, () => {
  matchesSnapshot(
    `
  import { graphql } from 'gatsby'

  export default () => (
    <div>{data.site.siteMetadata.title}</div>
  )

  export const siteMetaQuery = graphql\`
    fragment siteMetaQuery on RootQueryType {
      site {
        siteMetadata {
          title
        }
      }
    }
  \`

  export const query = graphql\`
     {
       ...siteMetaQuery
     }
  \`
  `
  )
})

it(`Handles closing StaticQuery tag`, () => {
  matchesSnapshot(`
  import React from 'react'
  import { graphql, StaticQuery } from 'gatsby'

  export default () => (
    <StaticQuery
      query={graphql\`{site { siteMetadata { title }}}\`}
    >
      {data => <div>{data.site.siteMetadata.title}</div>}
    </StaticQuery>
  )
  `)
})
