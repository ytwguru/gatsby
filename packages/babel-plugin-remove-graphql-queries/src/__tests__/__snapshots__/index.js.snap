// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handles closing StaticQuery tag 1`] = `
"import staticQueryData from \\"public/static/d/2626356014.json\\";
import React from 'react';
import { StaticQuery } from 'gatsby';
export default (() => React.createElement(StaticQuery, {
  query: \\"2626356014\\",
  data: staticQueryData
}, data => React.createElement(\\"div\\", null, data.site.siteMetadata.title)));"
`;

exports[`Leaves other graphql tags alone 1`] = `
"import React from 'react';
import { graphql } from 'relay';
export default (() => React.createElement(\\"div\\", null, data.site.siteMetadata.title));
export const query = graphql\`
     {
       site { siteMetadata { title }}
     }
  \`;"
`;

exports[`Removes all gatsby queries 1`] = `
"export default (() => React.createElement(\\"div\\", null, data.site.siteMetadata.title));
export const siteMetaQuery = \\"2673797374\\";
export const query = \\"2589775908\\";"
`;

exports[`Transforms queries in <StaticQuery> 1`] = `
"import staticQueryData from \\"public/static/d/2626356014.json\\";
import React from 'react';
import { StaticQuery } from 'gatsby';
export default (() => React.createElement(StaticQuery, {
  query: \\"2626356014\\",
  render: data => React.createElement(\\"div\\", null, data.site.siteMetadata.title),
  data: staticQueryData
}));"
`;

exports[`Transforms queries in page components 1`] = `"export const query = \\"3687030656\\";"`;

exports[`allows the global tag 1`] = `"export const query = \\"3687030656\\";"`;

exports[`distinguishes between the right tags 1`] = `
"const foo = styled('div')\`
     {
       \${foo}
     }
  \`;
const pulse = keyframes\`
    0% {
      transform: scale(1);
      animation-timing-function: ease-in;
    }
    25% {
      animation-timing-function: ease-out;
      transform: scale(1.05);
    }
    50% {
      transform: scale(1.12);
      animation-timing-function: ease-in;
    }
    to {
      transform: scale(1);
      animation-timing-function: ease-out;
    }
  \`;
export const query = \\"3687030656\\";"
`;

exports[`handles import aliasing 1`] = `"export const query = \\"3687030656\\";"`;

exports[`handles require 1`] = `"export const query = \\"3687030656\\";"`;

exports[`handles require alias 1`] = `"export const query = \\"3687030656\\";"`;

exports[`handles require namespace 1`] = `"export const query = \\"3687030656\\";"`;
